src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
```

Comment:
PrimathonAICodeReviewer

The `ControllerResponse` type and the `APIResponse.create` function look well-defined. However, there is room for improvement in handling the `data` and `error` properties more consistently.

```diff
+     if (error) {
+       response.error = error;
+     } else {
+       response.data = data || {};
```



<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->